src/relayclient/RelayClient.ts

```diff
@@ -212,28 +214,33 @@ export default class RelayClient {
   ): Promise<{ relayRequest: RelayRequest, relayMaxNonce: number, approvalData: PrefixedHexString, signature: PrefixedHexString, httpRequest: TmpRelayTransactionJsonRequest }> {
     const senderNonce = await this.contractInteractor.getSenderNonce(gsnTransactionDetails.from, gsnTransactionDetails.forwarder)
     const relayWorker = relayInfo.pingResponse.RelayServerAddress
-    const gasPrice = gsnTransactionDetails.gasPrice
-    const gasLimit = gsnTransactionDetails.gas
-    if (gasPrice == null || gasLimit == null) {
+    const gasPriceHex = gsnTransactionDetails.gasPrice
+    const gasLimitHex = gsnTransactionDetails.gas
+    if (gasPriceHex == null || gasLimitHex == null) {
       throw new Error('RelayClient internal exception. Gas price or gas limit still not calculated. Cannot happen.')
     }
+    if (gasPriceHex.indexOf('0x') !== 0 || gasLimitHex.indexOf('0x') !== 0) {
```

Comment:
`/0x\w+/.test(gasPriceHex)`
or, create `isHex()` method, which checks for non-null, prefix and character set)

(to be strict, the check is `0x[0-9a-fA-F]+`.test(..)`

Reply:
Regex are nice but hard to read. This is verbose but dead obvious what it does. There is absolutely no benefit in using regex here.

Reply:
AFAIK, single test is better than checking both for null and prefix (can have `isHex` too..)